src/utils.js

```diff
@@ -53,9 +58,222 @@ const jsonFlag = Flags.boolean({
+					`Please make sure the files ${path.basename(file)} and ${path.basename(
+						meshConfigName,
+					)} are in the same directory.`,
+				);
+			}
+		}
+	} catch (err) {
+		throw new Error(err.message);
+	}
+}
+
+/**
+ * Check if the files are of valid types .js, .json
+ *
+ * @param filesList List of files in mesh config
+ */
+function validateFileType(filesList) {
+	const filesWithInvalidTypes = [];
+
+	filesList.forEach(file => {
+		const extension = path.extname(file);
+		const isValidFileType = ['.js', '.json'].includes(extension);
+
+		if (!isValidFileType) {
+			filesWithInvalidTypes.push(file);
+		}
+	});
+
+	if (filesWithInvalidTypes.length) {
+		throw new Error(
+			`Mesh files must be JavaScript or JSON. Other file types are not supported. The following file(s) are invalid: ${path.basename(
+				filesWithInvalidTypes[0],
+			)}.`,
+		);
+	}
+}
+
+/**
+ * Validate the filenames
+ *
+ * @param filesList Files in sources, tranforms or additionalResolvers in the meshConfig
+ * @param data MeshConfig
+ */
+function validateFileName(filesList, data) {
+	const filesWithInvalidNames = [];
+
+	// Check if the file names are less than 25 characters
+	filesList.forEach(file => {
+		const fileName = path.basename(file);
+		if (fileName.length > 25) {
+			filesWithInvalidNames.push(fileName);
+		}
+	});
+
+	if (filesWithInvalidNames.length) {
+		throw new Error(
+			`Mesh file names must be less than 25 characters. The following file(s) are invalid: ${filesWithInvalidNames}.`,
+		);
+	}
+
+	// check if the the filePaths in the files array match
+	// the fileNames in sources, transforms or additionalResolvers
+	if (data.meshConfig.files) {
+		for (let i = 0; i < data.meshConfig.files.length; i++) {
+			if (filesList.indexOf(data.meshConfig.files[i].path) == -1) {
+				throw new Error(
+					`Please make sure the file names are matching in both places in meshConfig.`,
+				);
+			}
+		}
+	}
+}
+
+/**
+ * Append/override files to the files array in meshConfig
+ *
+ * @param data MeshConfig
+ * @param file File to append or override
+ * @param meshConfigName MeshConfig name
+ * @param index Append operation if index is -1, else override, it is the index where the override takes place
+ */
+function updateFilesArray(data, file, meshConfigName, index) {
+	try {
+		const readFileData = fs.readFileSync(
+			path.resolve(path.dirname(meshConfigName), file),
+			{ encoding: 'utf-8' },
+			err => {
+				if (err) {
+					throw new Error(err);
+				}
+			},
+		);
+
+		//data to be overridden or appended
+		const dataInFilesArray = jsmin(readFileData);
+
+		if (index >= 0) {
+			data.meshConfig.files[index] = {
+				path: `${file}`,
+				content: `${dataInFilesArray}`,
+			};
+		} else {
+			//if the files array does not exist
+			if (!data.meshConfig.files) {
+				data.meshConfig.files = [];
+			}
+
+			//if the files arrray exists, we append the file path and content in meshConfig
+			data.meshConfig.files.push({
+				path: `./${path.basename(file)}`,
```

Comment:
What happens if the file is in a different location than where the meshConfig is , like a parent directory?
I don’t think we should hardcode the path to the current directory where meshConfig isPlease make sure that if the file path is available, use that for setting files.path 
For example, 
```
"additionalResolvers": [
  "../additional-resolvers-2.js"
]
```

Reply:
If it is not a fully-qualified meshConfig, we are restricting the files to the same folder as mesh config.
To be checked with Revanth.

Reply:
Done